### PR TITLE
Add Game of Life exercise

### LINKED ISSUE #2790: game-of-life: implement practice exercise
The exercise `game-of-life` has not been implemented yet for the Java track.
The description of the exercise can be found in the [problem specification repository](https://github.com/exercism/problem-specifications/tree/main/exercises/game-of-life).

How to implement a new exercise for the Java track is described in detail in [CONTRIBUTING.md](https://github.com/exercism/java/blob/main/CONTRIBUTING.md). Please have a look there first before starting working on the exercise.
Also please make sure it is clear that you are currently working on this issue by asking to be assigned to it.

When opening an PR, please reference this issue using any of the [closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords).

In case you have any further questions, feel free to ask here.

### PR DESCRIPTION
**Pull request by akbatra567**
# pull request

* Add Game of life based on the instructions and data provided [here.](https://github.com/exercism/problem-specifications/tree/main/exercises/game-of-life)
* Closes #2790 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/settings.gradle [hunk 1]
```diff
@@ -158,3 +158,4 @@ include 'practice:wordy'
 include 'practice:yacht'
 include 'practice:zebra-puzzle'
 include 'practice:zipper'
+include 'practice:game-of-life'
```
Comment (kahgoh) [inline exercises/settings.gradle:161]: The list is alphabetically sorted. Could you please move this to keep the practice exercise list sorted?

File: config.json [hunk 2]
```diff
@@ -1791,6 +1791,17 @@
           "lists"
         ],
         "difficulty": 10
+      },
+      {
+        "slug": "game-of-life",
+        "name": "Conway's Game of Life",
+        "uuid": "a5f8aef3-9661-49c7-9eb3",
+        "practices": ["multi-dimensional-arrays"],
```
Comment (manumafe98) [inline config.json:1799]: Please leave `practices` empty as it's going to get treated in the following [issue](https://github.com/exercism/java/issues/2631)

File: config.json [hunk 3]
```diff
@@ -1791,6 +1791,17 @@
...
+        "prerequisites": [
+          "multi-dimensional-arrays",
```
Comment (manumafe98) [inline config.json:1801]: As there is no concept called multi dimensional arrays it should point to the arrays concept instead.

```suggestion
          "arrays",
```

File: exercises/practice/game-of-life/.meta/config.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "akbatra567"
+  ],
+  "files": {
+    "solution": [
+      "src/main/java/GameOfLife.java"
+    ],
+    "test": [
+      "src/test/java/GameOfLifeTest.java"
+    ],
+    "example": [".meta/src/reference/java/GameOfLife.java"],
```
Comment (manumafe98) [inline exercises/practice/game-of-life/.meta/config.json:12]: I would change `example` to the following to keep it consistent with our current formatting

```suggestion
    "example": [
      ".meta/src/reference/java/GameOfLife.java"
    ],
```

**Comment by manumafe98** [comment]
Really nice job @akbatra567! I would suggest to change the PR description with the keyword `closes` , so when the pr is done it closes the issue as well, here you can see an [example](https://github.com/exercism/java/pull/2773)

**Review by manumafe98 — CHANGES_REQUESTED** [review]

File: exercises/practice/game-of-life/.meta/config.json (hunk 4 above)
Comment (sanderploegsma) [reply exercises/practice/game-of-life/.meta/config.json:12]: I believe the Configlet also complains about this in the CI build 😉

File: exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
Comment (sanderploegsma) [inline exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties:3]: Other practice exercises are using Gradle wrapper version 8.7, so I was wondering how you created the wrapper. If you copied this from somewhere in this repo it means we missed something in the latest upgrades, which should be addressed.

If you created the wrapper yourself, please make sure to use the same version as the other exercises. This document explains how to update/sync Gradle versions: https://github.com/exercism/java/blob/main/reference/how-to-update-gradle.md

**Review by sanderploegsma — CHANGES_REQUESTED** [review]
This looks great!

File: exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (akbatra567) [reply exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties:3]: I followed the steps and the Contributing.md file, and copied `exercise-template`. This is coming from that. Will update the Gradle version.

Comment (sanderploegsma) [reply exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties:3]: Ah, so we missed the exercise template when updating the Gradle version. Good to know!

Perhaps you can also update the exercise template itself in this PR and add a note to the document on how to update Gradle that the template should be updated as well?

Comment (akbatra567) [reply exercises/practice/game-of-life/gradle/wrapper/gradle-wrapper.properties:3]: Sure, will update

**Review by kahgoh — APPROVED** [review]

**Review by manumafe98 — APPROVED** [review]

**Pull request merged**